pandas/tseries/offsets.py

```diff
@@ -1385,7 +1385,7 @@ def _end_apply_index(self, dtindex):
             roll = self.n
 
         base = (base_period + roll).to_timestamp(how='end')
-        return base + off
+        return base + off + Timedelta(1, 'ns') - Timedelta(1, 'D')
```

Comment:
hmm, might be better to actually add an offset here (e.g. a Day) which handles freq moves.

Reply:
Day() and Timedelta(days=1) should be equivalent here, no?  I'd rather use Timedelta, since Day is a less well-known beast.

Are there scenarios where a DST transition is traversed?

Reply:
I am thinking this will correctly handle the B freqs. its IS possible DST is traversed, but we prob don't have tests for this (though maybe)